### PR TITLE
[docs] added depends cross compile info

### LINKED ISSUE #7473: Depends system needs to figure more prominently in build docs
Some people have spent a lot of time figuring out how to build dependencies, or even how to cross-compile them, before realizing that the depends system makes this trivial.
This is because the documentation is a bit hidden, it is not linked from build-unix.md for example.

Also needs an example of cross-compiling, see here :https://github.com/bitcoin/bitcoin/issues/7471#issuecomment-180261245

### PR DESCRIPTION
**Pull request by accraze**
This PR contains additions to **doc/build-unix**:
- Added link to depends system readme
- Added cross compile info

Resolves #7473

### CODE REVIEW

File: doc/build-unix.md [hunk 1]
```diff
@@ -248,3 +248,27 @@ Additional Configure Flags
...
+installing the toolchain will be different.
+
+First install the toolchain:
+
+    sudo apt-get install g++-arm-linux-gnueabi
+
+To build executables for ARM:
+
+    cd depends
+    make -j10 HOST=arm-unknown-linux-gnueabihf NO_QT=1 # NO_WALLET=1 NO_UPNP=1 ...
```
Comment (btcdrak) [inline doc/build-unix.md]: Not sure I agree with `-j10` as a default for documentation. Better leave it out entirely as is the case for existing documentation.

**Comment by btcdrak** [comment]
barring my nit, utACK

File: doc/build-unix.md (hunk 1 above)
Comment (accraze) [inline doc/build-unix.md]: yeah @btcdrak  I can understand that .... do you also think I should get rid of `-j10` on line 270 as well?

Comment (btcdrak) [inline doc/build-unix.md]: @accraze Yes. Advanced users will add their own `-j` settings as appropriate.

**Comment by btcdrak** [comment]
utACK 7c99261

File: doc/build-unix.md [hunk 2]
```diff
@@ -248,3 +248,27 @@ Additional Configure Flags
...
+installing the toolchain will be different.
+
+First install the toolchain:
+
+    sudo apt-get install g++-arm-linux-gnueabi
+
+To build executables for ARM:
+
+    cd depends
+    make HOST=arm-unknown-linux-gnueabihf NO_QT=1 # NO_WALLET=1 NO_UPNP=1 ...
```
Comment (btcdrak) [inline doc/build-unix.md]: Should remove the need to remove the `...`

File: doc/build-unix.md [hunk 3]
```diff
@@ -248,3 +248,27 @@ Additional Configure Flags
...
+    cd ..
+    ./configure --prefix=$PWD/depends/arm-unknown-linux-gnueabihf  --enable-glibc-back-compat --enable-reduce-exports LDFLAGS=-static-libstdc++
+    make
+    scp src/bitcoind src/bitcoin-cli src/bitcoin-tx src/test/test_bitcoin $TARGET:
```
Comment (btcdrak) [inline doc/build-unix.md]: Not sure why there is an scp line here, seems othogonal.

Comment (accraze) [inline doc/build-unix.md]: you're right, it should be removed :+1:

Comment (MarcoFalke) [inline doc/build-unix.md]: Could remove this and then also squash the commits, please?

**Comment by btcdrak** [comment]
I got the following error running the make in `depends` folder

```
$ make HOST=arm-unknown-linux-gnueabihf NO_QT=1 # NO_WALLET=1 NO_UPNP=1
Building boost...
/home/drak/bitcoin/depends/work/build/arm-unknown-linux-gnueabihf/boost/1_59_0-16ec5a90819/tools/build/src/tools/gcc.jam:83: in gcc.init from module gcc
error: toolset gcc initialization:
error: provided command 'arm-unknown-linux-gnueabihf-g++' not found
error: initialized from /home/drak/bitcoin/depends/work/build/arm-unknown-linux-gnueabihf/boost/1_59_0-16ec5a90819/user-config.jam:1
/home/drak/bitcoin/depends/work/build/arm-unknown-linux-gnueabihf/boost/1_59_0-16ec5a90819/tools/build/src/build/toolset.jam:43: in toolset.using from module toolset
/home/drak/bitcoin/depends/work/build/arm-unknown-linux-gnueabihf/boost/1_59_0-16ec5a90819/tools/build/src/build/project.jam:1052: in using from module project-rules
/home/drak/bitcoin/depends/work/build/arm-unknown-linux-gnueabihf/boost/1_59_0-16ec5a90819/user-config.jam:1: in modules.load from module user-config
/home/drak/bitcoin/depends/work/build/arm-unknown-linux-gnueabihf/boost/1_59_0-16ec5a90819/tools/build/src/build-system.jam:249: in load-config from module build-system
/home/drak/bitcoin/depends/work/build/arm-unknown-linux-gnueabihf/boost/1_59_0-16ec5a90819/tools/build/src/build-system.jam:385: in load-configuration-files from module build-system
/home/drak/bitcoin/depends/work/build/arm-unknown-linux-gnueabihf/boost/1_59_0-16ec5a90819/tools/build/src/build-system.jam:524: in load from module build-system
/home/drak/bitcoin/depends/work/build/arm-unknown-linux-gnueabihf/boost/1_59_0-16ec5a90819/tools/build/src/kernel/modules.jam:295: in import from module modules
/home/drak/bitcoin/depends/work/build/arm-unknown-linux-gnueabihf/boost/1_59_0-16ec5a90819/tools/build/src/kernel/bootstrap.jam:139: in boost-build from module
/home/drak/bitcoin/depends/work/build/arm-unknown-linux-gnueabihf/boost/1_59_0-16ec5a90819/boost-build.jam:17: in module scope from module

funcs.mk:238: recipe for target '/home/drak/bitcoin/depends/work/build/arm-unknown-linux-gnueabihf/boost/1_59_0-16ec5a90819/./.stamp_built' failed
make: *** [/home/drak/bitcoin/depends/work/build/arm-unknown-linux-gnueabihf/boost/1_59_0-16ec5a90819/./.stamp_built] Error 1
```

File: doc/build-unix.md [hunk 4]
```diff
@@ -248,3 +248,26 @@ Additional Configure Flags
...
+installing the toolchain will be different.
+
+First install the toolchain:
+
+    sudo apt-get install g++-arm-linux-gnueabi
+
+To build executables for ARM:
+
+    cd depends
+    make HOST=arm-unknown-linux-gnueabihf NO_QT=1 # NO_WALLET=1 NO_UPNP=1
```
Comment (btcdrak) [inline doc/build-unix.md]: Please remove the comment `# NO_WALLET=1 NO_UPNP=1`

File: doc/build-unix.md [hunk 5]
```diff
@@ -248,3 +248,26 @@ Additional Configure Flags
...
+installing the toolchain will be different.
+
+First install the toolchain:
+
+    sudo apt-get install g++-arm-linux-gnueabi
+
+To build executables for ARM:
+
+    cd depends
+    make HOST=arm-unknown-linux-gnueabihf NO_QT=1
```
Comment (btcdrak) [inline doc/build-unix.md]: `-unknown` appears to be incorrect also.

Comment (accraze) [inline doc/build-unix.md]: yep just realized that.... it should be `arm-linux-gnueabihf`

File: doc/build-unix.md [hunk 6]
```diff
@@ -248,3 +248,26 @@ Additional Configure Flags
...
+
+Cross-compilation
+-------------------
+These steps can be performed on, for example, an Ubuntu VM. The depends system
+will also work on other Linux distributions, however the commands for
+installing the toolchain will be different.
+
+First install the toolchain:
+
+    sudo apt-get install g++-arm-linux-gnueabi
```
Comment (btcdrak) [inline doc/build-unix.md]: I believe this should be `g++-arm-linux-gnueabihf`

File: doc/build-unix.md [hunk 7]
```diff
@@ -248,3 +248,26 @@ Additional Configure Flags
...
+
+To build executables for ARM:
+
+    cd depends
+    make HOST=arm-linux-gnueabihf NO_QT=1
+    cd ..
+    ./configure --prefix=$PWD/depends/arm-unknown-linux-gnueabihf  --enable-glibc-back-compat --enable-reduce-exports LDFLAGS=-static-libstdc++
```
Comment (btcdrak) [inline doc/build-unix.md]: remove `-unknown` and the rogue space.

File: doc/build-unix.md [hunk 8]
```diff
@@ -248,3 +248,26 @@ Additional Configure Flags
...
+First install the toolchain:
+
+    sudo apt-get install g++-arm-linux-gnueabihf
+
+To build executables for ARM:
+
+    cd depends
+    make HOST=arm-linux-gnueabihf NO_QT=1
+    cd ..
+    ./configure --prefix=$PWD/depends/ar-linux-gnueabihf --enable-glibc-back-compat --enable-reduce-exports LDFLAGS=-static-libstdc++
```
Comment (btcdrak) [inline doc/build-unix.md]: typo `ar-` should be `arm-`

**Comment by btcdrak** [comment]
Tested ACK d82d94c

**Comment by jonasschnelli** [comment]
ACK d82d94c92dcd047ab465dad02c4727824cf1c730

File: doc/build-unix.md [hunk 9]
```diff
@@ -248,3 +248,26 @@ Additional Configure Flags
...
+
+
+
+Cross-compilation
+-------------------
+These steps can be performed on, for example, an Ubuntu VM. The depends system
+will also work on other Linux distributions, however the commands for
+installing the toolchain will be different.
+
+First install the toolchain:
```
Comment (paveljanik) [inline doc/build-unix.md]: You are installing the ARM specific toolchain before actually telling the reader that you'll be building for ARM.

Comment (petertodd) [inline doc/build-unix.md]: Just change the header of this section to "ARM Cross-compilation"

Comment (accraze) [inline doc/build-unix.md]: :+1: works for me

**Comment by btcdrak** [comment]
@laanwj Could this be backported to 0.12 please?

**Pull request merged**